subprojects/diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/DependencyInsightReportTaskIntegrationTest.groovy

```diff
@@ -1883,6 +1888,9 @@ project :impl
       org.gradle.usage               = java-api
       org.gradle.libraryelements     = jar (compatible with: classes)
       org.gradle.jvm.version         = $jvmVersion
+
+      Requested attributes not found in the selected variant:
+         org.gradle.jvm.environment     = standard-jvm
```

Comment:
Why is the attribute not added to consumable configurations?

Reply:
The attribute is a way to express that a variant has been optimised for a certain environment. By default, we should not make assumptions about this. If you publish a library through a "java-library" project, it is usually suited for all 'Java environment' without an optimisation. I think that's the case for most libraries today.

Also, if we would add the attribute to the consumables, we will change the defaults that are published in Gradle 7.x. Which can cause trouble if you add additional variants, and possibly also in combination with other plugins that add variants (e.g. Kotlin MPP).

I think in this case, we probably never want to set something by default, because of what I stated in the first paragraph. But we could revisit that decision later. For now I would start with just offering support for the attribute on the consuming side and then allow libraries to produce variants with the attribute if they really need it (like Guava).

Reply:
Ok, I think it makes sense to be careful. I was surprise because I interpreted the spec differently.